src/RpcClient/TransactionManager.cs

```diff
@@ -194,13 +180,18 @@ public TransactionManager AddWitness(UInt160 scriptHash, params object[] paramet
         /// <summary>
         /// Verify Witness count and add witnesses
         /// </summary>
-        public TransactionManager Sign()
+        /// <param name="snapshot">Snapshot required for Compute NetworkFee</param>
+        public TransactionManager Sign(SnapshotView snapshot)
         {
-            // Calculate NetworkFee
-            Tx.NetworkFee = CalculateNetworkFee();
-            var gasBalance = nep5API.BalanceOf(NativeContract.GAS.Hash, Tx.Sender);
-            if (gasBalance < Tx.SystemFee + Tx.NetworkFee)
-                throw new InvalidOperationException($"Insufficient GAS in address: {Tx.Sender.ToAddress()}");
+            if (snapshot != null)
```

Comment:
If we use `snapshot == null` we can set the networkFee manually with the `Tx` property